src/data_access/FirebaseAccessObject.java

```diff
@@ -66,4 +68,74 @@ public boolean existsByName(String newProjectName) {
         //TODO: add ways to check if newProjectName exists in db collection
         return true;
     }
+
+    public void addAnnouncement(CommonAnnouncement announcement) {
+        // Initialize Firestore if not already done
+        if (db == null) {
+            // Initialize Firestore
+        }
+
+        // Generate a unique ID for the announcement
+        String announcementId = UUID.randomUUID().toString();
+
+        // Convert LocalDateTime to a Firebase compatible format
+        String formattedCreationTime = announcement.getCreationTime()
+                .format(DateTimeFormatter.ISO_LOCAL_DATE_TIME);
+
+        // Create a Map to hold announcement data
+        Map<String, Object> announcementData = new HashMap<>();
+        announcementData.put("id", announcementId); // Add the generated ID
+        announcementData.put("title", announcement.getAnnouncementTitle());
+        announcementData.put("message", announcement.getMessage());
+        announcementData.put("creationTime", formattedCreationTime);
+        announcementData.put("author", announcement.getAuthor());
+
+        // Use the generated ID as the document ID in Firestore
+        ApiFuture<WriteResult> addedDocRef = db.collection("announcements").document(announcementId).set(announcementData);
+        // Handle completion of the future
+    }
+
+    @Override
+    public boolean deleteAnnouncement(String announcementId) {
+        // Assuming that the Firestore database has already been initialized in the constructor
+        // and db is your Firestore instance
+        try {
+            ApiFuture<WriteResult> writeResult = db.collection("announcements").document(announcementId).delete();
+            writeResult.get(); // This line throws InterruptedException or ExecutionException
+            return true; // Return true if deletion is successful
+        } catch (Exception e) {
+            e.printStackTrace();
+            return false; // Return false if an exception occurs
```

Comment:
instead of returning false, can't we throw some exception error to other classes (like interactor) and let that handle? (I'm not completely sure) 
If you look at my gmailDAO, the methods throw IOExceptions and google specific Exceptions. I am not sure which way is correct (if we can do both ways, which way should we agree?)